tracer/src/Datadog.Trace/Util/ThrowHelper.cs

```diff
@@ -0,0 +1,71 @@
+// <copyright file="ThrowHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace Datadog.Trace.Util
+{
+    internal class ThrowHelper
+    {
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
How come these are marked as `NoInlining`?

Reply:
Currently a method with throw doesn't get inlined, that's why we use the helper, to let the caller (with the non exceptional code) to be inlined, so I just wanted to annotate that with the attribute.

Surprisingly, affects the code gen: 
https://sharplab.io/#v2:EYLgHgbALANALiAhgZwLYB8ACAmAjAWAChMAGAAk1wDoAlAVwDs4BLVAUyoGEB7VAB2YAbNgCcAyqIBuzAMZtkAbiJFMAZgrYynMgG8iZA2X2G1FKGQCyuABTMmZZgEpdxw4eYBqD0sJu3ASRsnHz93LxDDAF9XI18TdUxzC2xbeycXONDPbxiAlODcsJzMsmiSmNNEskDUuAdnPRK3ZgAzMlsyAF5OshIGwtCAFQALEW4Ad2tHCNCy0IMBilwATmsAIgAJNkFBbjIAdW4RQQATNemYucMKhPN/fLT+pvc2ju7ep/m/OFGJsgY2OMyABRMByPgsbgMKYzPxXUKLSirTbbXYHI6nc6w+GxNwAbQsbB+3BO/n4gmshOJpPJAHkIcwocgqAA5bj+BiCOx2ADmjgAujczGQRmNJp95j8xf9ASCwWwGVCYZciJEgA

Try commenting the attribute :).

The ASM is very similar but with the attribute is still a bit shorter...

The weird thing to notice is the change from `je` to `jne` or `jz` to `jnz` . I don't know how this can affect the branch predictor. Any idea @kevingosse ?